File: OCH_Tao/ABC/A/087.js
const input = (require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n")).map(x => parseInt(x, 10));
let x = input[0];
x -= input[1];

while (input[2] <= x) {
  x -= input[2];
}
console.log(x);

File: OCH_Tao/ABC/B/087.py
A = int(input())
B = int(input())
C = int(input())
X = int(input())
l = []
for i in range(A+1):
  for j in range(B+1):
    for k in range(C+1):
      l.append(500*i+100*j+50*k)
print(len([i for i in l if i==X]))